src/components/Button/index.tsx

```diff
@@ -0,0 +1,168 @@
+import type { MouseEventHandler, ReactElement } from 'react'
+import { hexToCSSFilter } from 'hex-to-css-filter'
+import styled from '@emotion/styled'
+import type { Theme } from '@emotion/react'
+
+const ICON_SIZE = {
+  LARGE: 24,
+  SMALL: 16
+}
+export const BUTTON_STYLE_KEYS = {
+  GHOST: 'ghost',
+  OUTLINE: 'outline',
+  OUTLINE_DISABLED: 'outlineDisabled',
+  SOLID_DISABLED: 'solidDisabled',
+  SOLID_PRIMARY: 'solidPrimary',
+  SOLID_SUB: 'solidSub'
+}
+
+type ButtonStyle = typeof BUTTON_STYLE_KEYS[keyof typeof BUTTON_STYLE_KEYS]
+type ButtonSize = 'large' | 'medium' | 'small'
+export interface ButtonProps {
+  buttonStyle?: ButtonStyle
+  size?: ButtonSize
+  type?: 'button' | 'submit' | 'reset'
+  children: string
+  option?: string
+  iconUrl?: string
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+type StyledButtonProps = Pick<ButtonProps, 'buttonStyle' | 'size'>
+
+export const Button = ({
+  size,
+  buttonStyle,
+  type = 'button',
+  children,
+  option,
+  iconUrl,
+  onClick
+}: ButtonProps): ReactElement => {
+  const iconSize = size === 'small' ? ICON_SIZE.SMALL : ICON_SIZE.LARGE
+
+  return (
+    <StyledButton
+      buttonStyle={buttonStyle}
+      size={size}
+      type={type}
+      onClick={onClick}>
+      {iconUrl && (
+        <img alt="icon" height={iconSize} src={iconUrl} width={iconSize} />
+      )}
+      {children}
+      {option && <StyledButtonOption>{option}</StyledButtonOption>}
+    </StyledButton>
+  )
+}
+
+const StyledButton = styled.button<StyledButtonProps>`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border: none;
+  cursor: pointer;
+
+  ${({ theme }): string => theme.fonts.body02.bold}
+  ${({ theme, size = 'medium', buttonStyle = 'solidPrimary' }): string =>
+    renderButtonSizeStyle(theme, size, buttonStyle)}
+  ${({ theme, buttonStyle = 'solidPrimary' }): string =>
+    renderButtonColor(theme, buttonStyle as ButtonStyle)}
+
+  img {
+    padding-right: 4px;
+    filter: ${({ theme, buttonStyle = 'solidPrimary' }): string =>
```

Comment:
```suggestion
  ${({ theme, size, buttonStyle }): string =>
    renderButtonSizeStyle(theme, size, buttonStyle)}
  ${({ theme, buttonStyle }): string =>
    renderButtonColor(theme, buttonStyle)}

  img {
    padding-right: 4px;
    filter: ${({ theme, buttonStyle }): string =>
```
- 기본값을 컴포넌트쪽에서 받는 걸로 optional props에 대한 기본값을 더 명시적으로 확인할 수 있을 것 같습니다!
- styled props를 새롭게 정의하여 styled component 내부 기본 값, 타입 단언을 제거할 수 있을 것 같습니다!